
\begin{abstract}

新一代基因测序技术(NGS)的出现使得测序成本飞速下降，随之而来的是大量的短读
需要更快速准确的比对程序来处理。第一代基于散列表技术的序列比对算法如MAQ等
能够快速准确的完成比对工作，但其不支持gap比对的特性使得在短读序列(short reads)
过长导致indel出现频繁时，比对的精度也随之下降。另一方面，近年来压缩索引(BWT,CSA,FM-index)
领域的相关研究使得在内存中索引人类基因组这样的大规模序列成为可能。这导致
近年来出现了很多基于压缩索引的短读比对算法，如BWA等。本文提出了一种基于压
缩后缀数组和后向搜索实现的近似匹配的算法来实现短读比对，在比对时间和比对
精度上都取得了很好的效果。

本文提出并实现的基于压缩后缀数组的短读比对算法(CSAA)，采用了压缩后缀数组
的后向搜索来做近似匹配。通过引入搜索树，CSAA支持完全的gap比对。另一方面
CSAA在搜索树上使用了一种类似堆的优先堆数据结构，使得搜索树上的搜索空间大大
下降，而且每一次的搜索方向都能保证是最优的。最后结合罚分机制以及difference
距离，定义seed等方法，进一步降低搜索空间，提高CSAA的比对速度。

CSAA支持单端和双端序列比对，以Fastq格式输入，输出为标准的SAM(Sequence Alignment Map)
格式。CSAA还支持多线程比对，可以根据计算机的运算能力指定多个线程，以取得
数倍的加速优势。

\keywords{短读比对，序列比对，压缩索引，压缩后缀数组}
\end{abstract}

\begin{englishabstract}
Nowadays,decreasing cost and better accessibility of next generation sequencing methods
have produced a large amount of short reads whic is calling for the development of fast
and accurate read alignment programs.A first generation of hash-table based methods has
been developed,including MAQ,which is accurate,feature rich and fast enough to align short
reads from a single individual.However,MAQ does not support gapped alignment of longer reads
where indels may occur frequently.On the other hand,recent experimental studies on compressed
indexs(BWT,CSA,FM-index)have confirmed their practivality for indexing very long strings such
as human genome in the main memory,and many alignment method based on compressed index have
been developed,for example,BWA.In this paper we show how to build a software called CSAA that
exploits a CSA index of reference sequence,and perform well on alignment speed and accuration.

We proposed and implemented Compresses Suffix Array Alignment(CSAA),a new short read alignment
tool that is bases on backward search with compressed suffix array,to align short reads to a
large reference such as human genome.CSAA use a seach tree on multiple proximate sequences to
support mismatch and gapped alignment,CSAA also introduced a heap like structure to decrease
search space on seach tree.Finally,with the help of penalty strategy and seed,CSAA achieved
similar accuracy and faster speed than MAQ.

CSAA supports single-end and pair-end mapping with Fastq as input format and SAM(Sequence Alignment Map)
as output format.CSAA also support multi-thread running on a multi-core machine to get a faster
alignment speed.

\englishkeywords{short reads alignment, DNA sequence alignment, comressed index, compress suffix array}
\end{englishabstract}

